components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +132,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.where(column, OperatorType.LIKE, "%");
```

Comment:
Not confirmed hypothesis is that '' is in Oracle always treated as null, which means that COLUMN != '' is actually COLUMN != NULL and all comparisons against NULL are false. So, the way is to remove empty string from the query in case of Oracle and leave only IS/NOT NULL.